src/core/distributions/phylogenetics/tree/birthdeath/BirthDeathBurstProcess.cpp

```diff
@@ -135,9 +135,9 @@ double BirthDeathBurstProcess::computeLnProbabilityTimes( void )
         if ( n.isRoot() == false )
         {
             double parent_age = n.getParent().getAge();
-            if ( fabs(node_age - burst_time) > 1E-10 &&
+            if ( fabs(node_age - burst_time) > 1E-4 &&
```

Comment:
Again, in my opinion these comparison thresholds should not be hardcoded in place. I thought that's what functions in `RbMathLogic.h` were for? In general, though, I also question whether these precision-based equality comparisons are really always necessary.

Reply:
My tests showed that if a tree is read in from a file, there are a lot of rounding errors for branch lengths and thus node ages. This means that the events for a simulated tree written to a file didn't match when read in again. The problem with a global threshold is that unfortunately there is not one threshold that works in all situations, because the rounding has different impacts in different situations.